scipy/datasets/tests/test_data.py

```diff
@@ -0,0 +1,73 @@
+from scipy.datasets._registry import registry
+from scipy.datasets._fetchers import data
+from scipy.datasets import ascent, face, electrocardiogram
+from numpy.testing import assert_equal, assert_almost_equal, suppress_warnings
+import os
+import pytest
+
+
+data_dir = data.path
+
+# https://github.com/scipy/scipy/pull/15607#issuecomment-1176457275
+# TODO: Remove warning filter after next certifi release
+with suppress_warnings() as sup:
+    sup.filter(category=DeprecationWarning)
+    import pooch
+
+
+def _has_hash(path, expected_hash):
+    """Check if the provided path has the expected hash."""
+    if not os.path.exists(path):
+        return False
+    return pooch.file_hash(path) == expected_hash
+
+
+class TestDatasets:
+
+    @pytest.fixture(scope='module', autouse=True)
+    def test_download_all(self):
+        # This fixture requires INTERNET CONNECTION
+
+        # test_setup phase
+        for dataset in registry:
+            data.fetch(dataset)
+
+        yield
+
+        # test_teardown phase
+
+        # TODO: Clear all cache from test_download_all
```

Comment:
For the follow-up PR on this: this isn't exactly right, the test should not clear the cache, otherwise we're re-downloading all data each time someone runs the test suite locally. You can give it an `xslow` marker to do that, so it's only run very occasionally.

Reply:
That's a good idea and would be better to implement that marking the test as slow.